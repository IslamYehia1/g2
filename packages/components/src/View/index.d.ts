export * from '../../types/components/View';
