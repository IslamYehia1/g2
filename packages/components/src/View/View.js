import { connect } from '@wp-g2/context';
import { BaseView } from '@wp-g2/styles';
import { is, warning } from '@wp-g2/utils';
import React from 'react';

import { Text } from '../Text';

function View({ children, forwardedRef, ...props }) {
	warning(
		is.string(children),
		'@wp-g2/components',
		'View',
		`children prop should not be a string.`,
		`To render text, use the <Text /> component.`,
		`This instance of <View /> will be using <Text /> instead.`,
	);

	if (is.string(children)) {
		return (
			<Text {...props} ref={forwardedRef}>
				{children}
			</Text>
		);
	}

	return (
		<BaseView {...props} ref={forwardedRef}>
			{children}
		</BaseView>
	);
}
export default connect(View, 'View');
