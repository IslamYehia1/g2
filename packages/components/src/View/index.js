export { BaseView as View } from '../BaseView';
