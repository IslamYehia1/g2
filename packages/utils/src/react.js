import React from 'react';
import { is } from './is';

export function getValidChildren(children) {
	return React.Children.toArray(children).filter((child) =>
		React.isValidElement(child),
	);
}

export function assignRef(ref, value) {
	if (ref == null) return;

	if (is.function(ref)) {
		ref(value);
		return;
	}

	try {
		//@ts-ignore
		ref.current = value;
	} catch (error) {
		throw new Error(`Cannot assign value '${value}' to ref '${ref}'`);
	}
}

export function mergeRefs(...refs) {
	return (value) => {
		refs.forEach((ref) => assignRef(ref, value));
	};
}
