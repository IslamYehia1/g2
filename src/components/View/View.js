import { Box } from 'theme-ui';

const View = Box;

export default View;
