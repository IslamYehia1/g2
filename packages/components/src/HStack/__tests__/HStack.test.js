import { render } from '@testing-library/react';
import React from 'react';

import { View } from '../../View';
import { HStack } from '../index';

describe('props', () => {
	test('should render correctly', () => {
		const { container } = render(
			<HStack>
				<View />
				<View />
			</HStack>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	test('should render alignment', () => {
		const { container } = render(
			<HStack alignment="center">
				<View />
				<View />
			</HStack>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	test('should render spacing', () => {
		const { container } = render(
			<HStack spacing={5}>
				<View />
				<View />
			</HStack>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
