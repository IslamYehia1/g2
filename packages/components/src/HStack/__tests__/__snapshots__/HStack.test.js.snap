// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render alignment 1`] = `
<div
  class="wp-components-view wp-components-flex wp-components-h-stack css-vqn2g8"
  data-g2-c16t="true"
  data-g2-component="HStack"
>
  <div
    class="wp-components-flex-item css-eqsycl"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <div
      class="wp-components-view css-1oanc4k"
      data-g2-c16t="true"
      data-g2-component="View"
    />
  </div>
  <div
    class="wp-components-flex-item css-tfelz2"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <div
      class="wp-components-view css-1oanc4k"
      data-g2-c16t="true"
      data-g2-component="View"
    />
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
<div
  class="wp-components-view wp-components-flex wp-components-h-stack css-1pg3rlk"
  data-g2-c16t="true"
  data-g2-component="HStack"
>
  <div
    class="wp-components-flex-item css-eqsycl"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <div
      class="wp-components-view css-1oanc4k"
      data-g2-c16t="true"
      data-g2-component="View"
    />
  </div>
  <div
    class="wp-components-flex-item css-tfelz2"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <div
      class="wp-components-view css-1oanc4k"
      data-g2-c16t="true"
      data-g2-component="View"
    />
  </div>
</div>
`;

exports[`props should render spacing 1`] = `
<div
  class="wp-components-view wp-components-flex wp-components-h-stack css-1pg3rlk"
  data-g2-c16t="true"
  data-g2-component="HStack"
>
  <div
    class="wp-components-flex-item css-h5jpv5"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <div
      class="wp-components-view css-1oanc4k"
      data-g2-c16t="true"
      data-g2-component="View"
    />
  </div>
  <div
    class="wp-components-flex-item css-tfelz2"
    data-g2-c16t="true"
    data-g2-component="FlexItem"
  >
    <div
      class="wp-components-view css-1oanc4k"
      data-g2-c16t="true"
      data-g2-component="View"
    />
  </div>
</div>
`;
